consensus/src/experimental/execution_phase.rs

```diff
@@ -1,121 +1,118 @@
-    ) -> anyhow::Result<()> {
-        // reset the execution phase
-
-        // notify the commit phase
-        let (tx, rx) = oneshot::channel::<ResetAck>();
-        self.commit_phase_reset_event_tx.send(tx).await?;
-        rx.await?;
-
-        // exhaust the executor channel
-        while self.executor_channel_rx.next().now_or_never().is_some() {}
+    pub async fn process(&self, item: ExecutionIncomingType) -> ExecutionOutgoingType {
+        let ExecutionIncomingType { blocks } = item;
 
-        // activate the callback
-        reset_event_callback
-            .send(reset_ack_new())
-            .map_err(|_| Error::ResetDropped)?;
+        // execute the blocks with execution_correctness_client
+        let out_item = blocks
+            .into_iter()
+            .map(|b| {
+                let state_compute_result = self
+                    .execution_proxy
+                    .compute(&b, b.parent_id())
+                    .map_err(|e| ExecutionPhaseError {
+                        block_id: b.id(),
+                        error: e,
+                    })?;
+                Ok(ExecutedBlock::new(b, state_compute_result))
+            })
+            .collect::<Result<Vec<ExecutedBlock>, ExecutionPhaseError>>();
 
-        Ok(())
+        ExecutionOutgoingType { inner: out_item }
     }
 
     pub async fn start(mut self) {
         // main loop
         loop {
             select! {
-                ExecutionChannelType(vecblock, ledger_info, callback) = self.executor_channel_rx.select_next_some() => {
-                    // execute the blocks with execution_correctness_client
-                    let executed_blocks: Vec<ExecutedBlock> = vecblock
-                        .into_iter()
-                        .map(|b| {
-                            let state_compute_result =
-                                self.execution_proxy.compute(&b, b.parent_id()).unwrap();
-                            ExecutedBlock::new(b, state_compute_result)
-                        })
-                        .collect();
-                    // TODO: add error handling. Err(Error::BlockNotFound(parent_block_id))
-
-                    // pass the executed blocks into the commit phase
-                    self.commit_channel_tx
-                        .send(CommitChannelType(executed_blocks, ledger_info, callback))
-                        .await
-                        .map_err(|e| ExecutionError::InternalError {
-                            error: e.to_string(),
-                        })
-                        .unwrap();
-                }
-                reset_event_callback = self.reset_event_channel_rx.select_next_some() => {
-                    self.process_reset_event(reset_event_callback).await.map_err(|e| ExecutionError::InternalError {
-                        error: e.to_string(),
-                    })
-                    .unwrap();
+                in_item = self.rx.select_next_some() => {
```

Comment:
for a single channel, don't need the select?

Reply:
resolved